steward/src/allocation.rs

```diff
@@ -50,114 +35,128 @@ pub async fn get_connections() -> Result<Connections, Error> {
             val,
             Contact::new(&config.cosmos.grpc, timeout, &config.cosmos.prefix)?,
         ),
-        Err(e) => {
-            error!(
-                "Failed to access Cosmos gRPC with {:?} and create connections",
-                e
-            );
-            return Err(Error::from(e));
+        Err(err) => {
+            error!("failed to create allocation client: {}", err);
+            return Err(Error::from(err));
         }
     };
     Ok(Connections { grpc, contact })
 }
 
 pub async fn decide_rebalance(
     tick_range: Vec<somm::TickRange>,
-    cellar_address: H160,
+    cellar_address: String,
 ) -> Result<(), Error> {
-    if std::env::var("CELLAR_DRY_RUN").expect("Expect CELLAR_DRY_RUN var") == "TRUE" {
-        return Ok(());
-    } else {
-        let mut connections = get_connections().await?;
-        let (contact, grpc_client) = (connections.contact, &mut connections.grpc);
+    debug!("deciding rebalance for cellar address {}", cellar_address);
+    let mut connections = get_connections().await?;
+    let (contact, grpc_client) = (connections.contact, &mut connections.grpc);
 
-        let eth_gas_price = match cellars::get_gas_price().await {
-            Ok(gp) => gp,
-            Err(err) => {
-                error!("Failed to get cellar gas price: {:?}", err);
-                // TO-DO handle this better
-                0.into()
-            }
-        };
-        let config = APP.config();
-        let allocation = to_allocation(
-            tick_range,
-            format_eth_address(cellar_address),
-            eth_gas_price.as_u64(),
-        );
+    debug!("getting eth gas price");
+    let eth_gas_price = match cellars::get_gas_price().await {
+        Ok(gp) => {
+            debug!("eth gas price is {}", gp);
+            gp
+        },
+        Err(err) => {
+            error!("failed to get cellar gas price: {}", err);
+            // TO-DO handle this better
+            0.into()
+        }
+    };
+    let config = APP.config();
 
-        let name = &config.keys.rebalancer_key;
-        let cosmos_key = config.load_deep_space_key(name.clone());
-        let delegate_cosmos_address = cosmos_key.to_address(&contact.get_prefix())?;
+    debug!("loading the delegate (orchestrator) key and address from config");
+    let name = &config.keys.rebalancer_key;
+    let cosmos_delegate_key = config.load_deep_space_key(name.clone());
+    let cosmos_delegate_address = cosmos_delegate_key.to_address(&config.cosmos.prefix)?;
 
-        let cosmos_gas_price = config.cosmos.gas_price.as_tuple();
-        let fee = Coin {
-            denom: cosmos_gas_price.1,
-            amount: 0u32.into(),
-        };
+    debug!("querying the validator address based on the associated delegate address {}", cosmos_delegate_address);
+    let mut gravity_client = GravityQueryClient::connect(config.cosmos.grpc.clone()).await?;
```

Comment:
I don't think this needs to be mutable. Would it make sense to add it to `get_connections()`?

Reply:
It does; something about the `delegate_keys_by_orchestrator()` method called in the following method in `utils` requires it.

Reply:
This call borrows `client` as mutable: https://github.com/PeggyJV/steward/blob/7533448a1505dbfb8c219b7f884c2e08cf257eb8/steward/src/utils.rs#L30
signature
https://github.com/PeggyJV/gravity-bridge/blob/19ba1cc74d0fdfd7ba9f3cfcdad354c7259030bc/orchestrator/gravity_proto/src/prost/gravity.v1.rs#L1263-L1266